runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpInfoBox.java

```diff
@@ -231,6 +231,12 @@ public void popupMenuCanceled(PopupMenuEvent popupMenuEvent)
 		add(container, BorderLayout.NORTH);
 	}
 
+	void setInfoBoxPosition(int position)
+	{
+		panel.setComponentZOrder(this, position);
+		panel.revalidate();
```

Comment:
Move this to the end of `XpPanel#sortByXp()` so it is called only once per sort instead of once per box. I've tested this and can confirm the UI is visually updated the same way with that change in place.